examples/example-metal-kernel.c

```diff
@@ -20,26 +20,26 @@ void exp_elemwise(
       "out[elem] = metal::exp(tmp);";
   mlx_vector_string input_names = mlx_vector_string_new_value("inp");
   mlx_vector_string output_names = mlx_vector_string_new_value("out");
-  mlx_fast_metal_kernel kernel = mlx_fast_metal_kernel_new(
+  mlx_fast_custom_kernel kernel = mlx_fast_custom_kernel_new_metal(
       "myexp", input_names, output_names, source, "", true, false);
 
-  mlx_fast_metal_kernel_config config = mlx_fast_metal_kernel_config_new();
+  mlx_fast_custom_kernel_config config = mlx_fast_custom_kernel_config_new();
```

Comment:
Just to be sure I understand. Metal uses `mlx_fast_custom_kernel_new_metal` and CUDA uses `mlx_fast_custom_kernel_new_cuda`. Then they share the other config modification functions and free functions?

I guess an alternative would be to separate everythin g and drop `custom`:
- For metal `mlx_fast_metal_kernel_new` and `mlx_fast_metal_kernel_config_...`
- For CUDA `mlx_fast_cuda_kernel_new` and `mlx_fast_cuda_kernel_config_...`

I think the first is fine.. unless we think the configs might change for different back-ends. Maybe @angeloskath has a thought on that.